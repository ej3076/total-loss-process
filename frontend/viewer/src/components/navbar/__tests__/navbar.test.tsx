import React from 'react';
import { MemoryRouter } from 'react-router';
import { create } from 'react-test-renderer';

import Navbar from '../';

describe('<Navbar />', () => {
  it('should render light mode correctly', () => {
    const tree = create(
      <MemoryRouter>
        <Navbar mode="" toggleMode={jest.fn()} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render dark mode correctly', () => {
    const tree = create(
      <MemoryRouter>
        <Navbar mode="DARK_MODE" toggleMode={jest.fn()} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
