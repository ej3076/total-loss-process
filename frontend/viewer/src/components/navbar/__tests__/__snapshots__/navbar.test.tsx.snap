// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render dark mode correctly 1`] = `
<div
  className="bp3-navbar"
>
  <div
    className="container"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <a
        className="bp3-button bp3-large bp3-minimal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="bp3-button-text"
        >
          <strong>
            ClaimTrack
          </strong>
        </span>
      </a>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="bp3-button bp3-minimal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="bp3-icon bp3-icon-feed"
          icon="feed"
        >
          <svg
            data-icon="feed"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              feed
            </desc>
            <path
              d="M1.99 11.99c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.89-2-2-2zm1-4c-.55 0-1 .45-1 1s.45 1 1 1c1.66 0 3 1.34 3 3 0 .55.45 1 1 1s1-.45 1-1c0-2.76-2.24-5-5-5zm0-4c-.55 0-1 .45-1 1s.45 1 1 1c3.87 0 7 3.13 7 7 0 .55.45 1 1 1s1-.45 1-1a9 9 0 0 0-9-9zm0-4c-.55 0-1 .45-1 1s.45 1 1 1c6.08 0 11 4.92 11 11 0 .55.45 1 1 1s1-.45 1-1c0-7.18-5.82-13-13-13z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Feed
        </span>
      </a>
      <a
        className="bp3-button bp3-minimal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="bp3-icon bp3-icon-document"
          icon="document"
        >
          <svg
            data-icon="document"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              document
            </desc>
            <path
              d="M9 0H3c-.55 0-1 .45-1 1v14c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V5L9 0zm3 14H4V2h4v4h4v8z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Claims
        </span>
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <button
        className="bp3-button bp3-minimal"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        title="Light mode"
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-flash"
          icon="flash"
        >
          <svg
            data-icon="flash"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              flash
            </desc>
            <path
              d="M4 8c0-.55-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1h2c.55 0 1-.45 1-1zm4-4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1S7 .45 7 1v2c0 .55.45 1 1 1zM3.79 5.21a1.003 1.003 0 0 0 1.42-1.42l-1.5-1.5a1.003 1.003 0 0 0-1.42 1.42l1.5 1.5zm.71 5.29c-.28 0-.53.11-.71.29l-1.5 1.5a1.003 1.003 0 0 0 1.42 1.42l1.5-1.5a1.003 1.003 0 0 0-.71-1.71zm7-5c.28 0 .53-.11.71-.29l1.5-1.5a1.003 1.003 0 0 0-1.42-1.42l-1.5 1.5a1.003 1.003 0 0 0 .71 1.71zm.71 5.29a1.003 1.003 0 0 0-1.42 1.42l1.5 1.5a1.003 1.003 0 0 0 1.42-1.42l-1.5-1.5zM15 7h-2c-.55 0-1 .45-1 1s.45 1 1 1h2c.55 0 1-.45 1-1s-.45-1-1-1zM8 5C6.34 5 5 6.34 5 8s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3zm0 4c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm0 3c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<Navbar /> should render light mode correctly 1`] = `
<div
  className="bp3-navbar"
>
  <div
    className="container"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <a
        className="bp3-button bp3-large bp3-minimal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="bp3-button-text"
        >
          <strong>
            ClaimTrack
          </strong>
        </span>
      </a>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        className="bp3-button bp3-minimal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="bp3-icon bp3-icon-feed"
          icon="feed"
        >
          <svg
            data-icon="feed"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              feed
            </desc>
            <path
              d="M1.99 11.99c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.89-2-2-2zm1-4c-.55 0-1 .45-1 1s.45 1 1 1c1.66 0 3 1.34 3 3 0 .55.45 1 1 1s1-.45 1-1c0-2.76-2.24-5-5-5zm0-4c-.55 0-1 .45-1 1s.45 1 1 1c3.87 0 7 3.13 7 7 0 .55.45 1 1 1s1-.45 1-1a9 9 0 0 0-9-9zm0-4c-.55 0-1 .45-1 1s.45 1 1 1c6.08 0 11 4.92 11 11 0 .55.45 1 1 1s1-.45 1-1c0-7.18-5.82-13-13-13z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Feed
        </span>
      </a>
      <a
        className="bp3-button bp3-minimal"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="bp3-icon bp3-icon-document"
          icon="document"
        >
          <svg
            data-icon="document"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              document
            </desc>
            <path
              d="M9 0H3c-.55 0-1 .45-1 1v14c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V5L9 0zm3 14H4V2h4v4h4v8z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Claims
        </span>
      </a>
      <div
        className="bp3-navbar-divider"
      />
      <button
        className="bp3-button bp3-minimal"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        title="Dark mode"
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-moon"
          icon="moon"
        >
          <svg
            data-icon="moon"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              moon
            </desc>
            <path
              d="M15 11.38A7.835 7.835 0 0 1 7.85 16C3.51 16 0 12.49 0 8.15 0 4.97 1.89 2.23 4.62 1c-.45.99-.7 2.08-.7 3.23a7.85 7.85 0 0 0 7.85 7.85c1.15 0 2.24-.25 3.23-.7z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
